src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerLimits.cs

```diff
@@ -143,20 +143,20 @@ public int MaxRequestHeaderCount
         }
 
         /// <summary>
-        /// Gets or sets the keep-alive timeout.
+        /// Gets or sets the connection timeout.
         /// </summary>
         /// <remarks>
         /// Defaults to 2 minutes. Timeout granularity is in seconds. Sub-second values will be rounded to the next second.
         /// </remarks>
-        public TimeSpan KeepAliveTimeout
+        public TimeSpan ConnectionTimeout
```

Comment:
I would rather keep this a timeout for only time spent between request. Having this timeout trigger because an app takes a long time doing business logic before writing a response or completing seems bad to me (even if it would never kill the connection mid request or response).